### PR TITLE
[WebProfiler] Disallow viewing dot-files in Profiler

### PR DESCRIPTION
**Pull request by curry684**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

The file viewer in the profiler should not open files that were specifically intended to be hidden, like specifically .env files, but similarly files like .htaccess that might expose server configuration knowledge.

Added tests validating both the new and old behavior.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
:-1: as explained on another issue: the profiler is already exposing sensitive information (eg. cookies)

The policy is to *not* consider anything done by the profiler as a security issue. Opening dot files is no exception IMHO.

> Based on 3.2 as it's a security issue

There is a BIG issue here: if this where to be a valid security issue, the public disclosure would be BAD.
Please consider the policy here: https://symfony.com/security
TL;DR: security issue must be reported on security@symfony.com ONLY.
Thanks :)

**Comment by curry684** [comment]
Responsible disclosure is in my opinion only required for exploits usable right now on active sites. This specific issue is a privilege escalation loophole that causes *more* problems after another attack, and it was already being discussed in other issues.

> The policy is to not consider anything done by the profiler as a security issue.

Well if that's the case I heartily disagree with the policy. Yes, access to the profiler should be considered an absolute and complete security breach of the **site it is part of**. No, it should not compromise the entire server or network. So by no means should the profiler be allowed to:
- Disclose files with absolute paths like `/env/passwd`
- Access potentially highly sensitive files on the host, like `.composer/auth.json`, or `.docker/config.json`, or `.ssh/id_rsa`, which incidentally are all in dot-folders for good reason
- Download my entire `.git` folder with commit histories et al, disclosing personal information and email addresses of my employees

The file viewer itself was already guarded against directory walking, with this PR this stretches to other files that should never be relevant through a web interface.

**Review by nicolas-grekas — APPROVED** [review]
ok, not a security issue, but considering as a bug is fine to me

**Comment by nicolas-grekas** [comment]
> Responsible disclosure is in my opinion only required for exploits usable right now on active sites

The issue is by disclosing publicly, this makes it impossible for anyone else to disagree, as it's public.
Please reconsider next time :)

File: src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php [hunk 1]
```diff
@@ -46,6 +47,40 @@ public function getEmptyTokenCases()
...
+        $twig = $this->getMockBuilder('Twig\Environment')->disableOriginalConstructor()->getMock();
+        $profiler = $this
+            ->getMockBuilder('Symfony\Component\HttpKernel\Profiler\Profiler')
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $controller = new ProfilerController($urlGenerator, $profiler, $twig, array(), 'bottom', null, __DIR__.'/../..');
+
+        if (!$isAllowed) {
+            $this->expectException(NotFoundHttpException::class);
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php]: fails on travis

Comment (curry684) [reply src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/ProfilerControllerTest.php]: I hate PHP 5.5. Should be fixed now.

**Comment by fabpot** [comment]
Thank you @curry684.
